File: contest/LeetCode_Weekly_Contest_201/Find_Kth_Bit_in_Nth_Binary_String.js
/**
 * @param {number} n
 * @param {number} k
 * @return {character}
 */
var findKthBit = function(n, k) {
    k--;
    n--;
    var ret = [0];
    for (var i  = 0; i < n; ++i) {
        var b = [];
        for (var j = ret.length-1, id = 0; j >= 0; --j, id++) {
            b[id] = 1 - ret[j];
        }
        ret = ret.concat([1], b);
    }
    
    return ret[k].toString();
};

File: contest/LeetCode_Weekly_Contest_201/Find_Kth_Bit_in_Nth_Binary_String.py
class Solution(object):
    def findKthBit(self, n, k):
        ret = [0]
        n -= 1
        k -= 1
        
        for i in xrange(0, n):
            b = map(lambda c: 0 if c else 1, ret[-1::-1])
            ret.append(1)
            ret.extend(b)
        return str(ret[k])
        
